src/P151_Honeywell_pressure.ino

```diff
@@ -0,0 +1,162 @@
+#include "_Plugin_Helper.h"
+#ifdef USES_P151
+
+// #######################################################################################################
+// ####################### Plugin 151 Honeywell Digital Output Pressure Sensors ##########################
+// #######################################################################################################
+
+
+# define PLUGIN_151
+# define PLUGIN_ID_151         151
+# define PLUGIN_NAME_151       "Environment - I2C Honeywell Pressure"
+# define PLUGIN_VALUENAME1_151 "Pressure"
+# define PLUGIN_VALUENAME2_151 "Temperature"
+
+# include "./src/PluginStructs/P151_data_struct.h"
+
+boolean Plugin_151(uint8_t function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number           = PLUGIN_ID_151;
+      Device[deviceCount].Type               = DEVICE_TYPE_I2C;
+      Device[deviceCount].VType              = Sensor_VType::SENSOR_TYPE_DUAL;
+      Device[deviceCount].Ports              = 0;
+      Device[deviceCount].PullUpOption       = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption      = true;
+      Device[deviceCount].ValueCount         = 2;
+      Device[deviceCount].SendDataOption     = true;
+      Device[deviceCount].TimerOption        = false;
+      Device[deviceCount].TimerOptional      = true;
+      Device[deviceCount].GlobalSyncOption   = true;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_151);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_151));
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_151));
+      break;
+    }
+
+    case PLUGIN_I2C_HAS_ADDRESS:
+    case PLUGIN_WEBFORM_SHOW_I2C_PARAMS:
+    {
+      // 8 bit I2C address notation in datasheet,
+      // thus have to shift all options 1 bit to the right.
+      // default: 40 (28 hex).
+      // Other available standard addresses are:
+      // 56 (38 hex)
+      // 72 (48 hex)
+      // 88 (58 hex)
+      // 104 (68 hex)
+      // 120 (78 hex)
+      // 136 (88 hex)
+      // 152 (98 hex)
+      const uint8_t i2cAddressValues[] = {
+        0x28 >> 1,
+        0x38 >> 1,
+        0x48 >> 1,
+        0x58 >> 1,
+        0x68 >> 1,
+        0x78 >> 1,
+        0x88 >> 1,
+        0x98 >> 1
+      };
+      constexpr size_t addrLength = sizeof(i2cAddressValues) / sizeof(uint8_t);
```

Comment:
I think a distinction between `PLUGIN_I2C_HAS_ADDRESS` and `PLUGIN_WEBFORM_SHOW_I2C_PARAMS` has to be added here to make the device appear in the I2C Scanner.

Reply:
Check!
Will look into it.
Right now the state of this PR is: it compiled :)